packages/client/src/views/SysAdmin/Config/Application/Application.tsx

```diff
@@ -516,41 +498,44 @@ function DeathTabContent({
   )
 }
 
-class ApplicationConfigComponent extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      activeTabId: TabId.GENERAL,
-      changeModalName: EMPTY_STRING,
-      showNotification: false,
-      notificationStatus: NOTIFICATION_TYPE.IN_PROGRESS,
-      notificationMessages: EMPTY_STRING
-    }
-  }
+function ApplicationConfigComponent() {
+  const intl = useIntl()
+  const offlineCountryConfiguration = useSelector((store: IStoreState) =>
+    getOfflineData(store)
+  )
+  const [activeTabId, setActiveTabId] = React.useState(TabId.GENERAL)
+  const [changeModalName, setModalName] = React.useState(EMPTY_STRING)
+  const [showNotification, setShowNotification] = React.useState(false)
+  const [notificationStatus, setNotificationStatus] = React.useState(
+    NOTIFICATION_TYPE.IN_PROGRESS
+  )
+  const [notificationMessages, setNotificationMessages] =
+    React.useState(EMPTY_STRING)
 
-  changeValue = (notificationStatus: NOTIFICATION_TYPE, messages: string) => {
+  const changeValue = (
+    notificationStatus: NOTIFICATION_TYPE,
+    messages: string
+  ) => {
     if (notificationStatus !== NOTIFICATION_TYPE.ERROR) {
-      this.toggleConfigModal()
-      this.setState({
-        showNotification: true,
-        notificationStatus: notificationStatus,
-        notificationMessages: messages
-      })
+      toggleConfigModal()
+      setShowNotification(true)
+      setNotificationStatus(notificationStatus)
+      setNotificationMessages(messages)
     }
   }
 
-  changeTab(id: string) {
-    this.setState({ activeTabId: id })
+  const changeTab = (id: TabId) => {
+    setActiveTabId(id)
   }
 
-  toggleConfigModal = () => {
-    if (this.state.changeModalName) {
-      this.setState({ changeModalName: '' })
+  const toggleConfigModal = () => {
+    if (changeModalName) {
+      setModalName('')
     }
-    return !!!this.state.changeModalName ? false : true
+    return !!!changeModalName ? false : true
```

Comment:
This should be equivalent

```suggestion
    return Boolean(changeModalName)
```
as
```ts
!!!"s" ? false : true
// true
!!!"" ? false : true
// false
```